File: config/src/main/java/org/wcl/mfn/config/ui/common/PageTitleConfig.java
package org.wcl.mfn.config.ui.common;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.*;

@Configuration
@PropertySource("classpath:ui/common/page-title.properties")
public class PageTitleConfig {
    @Value("${page.home.title}")
    private String homePageTitle;

    @Value("${page.contract-calculator.title}")
    private String contractCalculatorPageTitle;

    public String homePageTitle() {
        return homePageTitle;
    }

    public String contractCalculatorPageTitle() {
        return contractCalculatorPageTitle;
    }
}

File: config/src/main/kotlin/org/wcl/mfn/config/ui/common/PageTitleConfig.kt
package org.wcl.mfn.config.ui.common

import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.*

@Configuration
@PropertySource("classpath:ui/common/page-title.properties")
open class PageTitleConfig {
    @Value("\${page.home.title}")
    private val homePageTitle: String? = null

    @Value("\${page.contract-calculator.title}")
    private val contractCalculatorPageTitle: String? = null

    fun homePageTitle(): String? {
        return homePageTitle
    }

    fun contractCalculatorPageTitle(): String? {
        return contractCalculatorPageTitle
    }
}
